### ISSUE TITLE
Something appears to corrupt the overworld save

### ISSUE THREAD

**Issue by smartboyathome** [GlowstoneMC/Glowstone#620]
I was doing some testing of my own code and just flying around the attached world when I started to encounter some invalid length exceptions from the reader like so:

```
2017/12/17 21:05:35 [SEVERE] Error while loading chunk (6,20)
java.io.IOException: Invalid length: 0 <= 0 
	at net.glowstone.io.anvil.RegionFile.getChunkDataInputStream(RegionFile.java:232)
	at net.glowstone.io.anvil.AnvilChunkIoService.read(AnvilChunkIoService.java:60)
	at net.glowstone.chunk.ChunkManager.loadChunk(ChunkManager.java:150)
	at net.glowstone.chunk.GlowChunk.load(GlowChunk.java:200)
	at net.glowstone.generator.populators.StructurePopulator.populate(StructurePopulator.java:35)
	at net.glowstone.chunk.ChunkManager.populateChunk(ChunkManager.java:235)
	at net.glowstone.chunk.ChunkManager.forcePopulation(ChunkManager.java:249)
	at net.glowstone.entity.GlowPlayer.lambda$streamBlocks$6(GlowPlayer.java:952)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at net.glowstone.entity.GlowPlayer.streamBlocks(GlowPlayer.java:951)
	at net.glowstone.entity.GlowPlayer.pulse(GlowPlayer.java:787)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.LinkedList$LLSpliterator.forEachRemaining(LinkedList.java:1235)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at net.glowstone.GlowWorld.pulsePlayers(GlowWorld.java:581)
	at net.glowstone.GlowWorld.pulse(GlowWorld.java:461)
	at net.glowstone.scheduler.WorldScheduler$WorldThread.run(WorldScheduler.java:142)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

I switched over to the dev branch and it appears to happen there as well. I attached [the world](https://github.com/GlowstoneMC/Glowstone/files/1568536/corrupt-world.zip) that is causing this, including the configs and logs. I think that this is being caused by some problems with saving, and subsequently reading back, structures, but I haven't had enough time to properly diagnose this.

**Comment by aramperes** [comment]
I haven't been able to reproduce this with the latest dev builds on the same seed (`-6908717096376827460`). What build were you using at the time?

**Comment by Pr0methean** [comment]
Is it possible this was a result of overworld chunks being interleaved with chunks from other dimensions, and thus fixed when #630 was fixed?

**Comment by aramperes** [comment]
I assume 783d5d353802c42d7ff2bd43c6dd4be9d8f459ac fixed this issue
